android/app/src/main/java/com/microsoft/codepush/react/CodePushNativeModule.java

```diff
@@ -240,19 +245,20 @@ public void dispatchDownloadProgressEvent() {
 
     @ReactMethod
     public void getConfiguration(Promise promise) {
-        Activity currentActivity = getCurrentActivity();
         WritableNativeMap configMap = new WritableNativeMap();
         configMap.putString("appVersion", mCodePush.getAppVersion());
+        configMap.putString("clientUniqueId", mClientUniqueId);
         configMap.putString("deploymentKey", mCodePush.getDeploymentKey());
         configMap.putString("serverUrl", mCodePush.getServerUrl());
-        configMap.putString("clientUniqueId",
-                Settings.Secure.getString(currentActivity.getContentResolver(),
-                        android.provider.Settings.Secure.ANDROID_ID));
-        String binaryHash = CodePushUpdateUtils.getHashForBinaryContents(currentActivity, mCodePush.isDebugMode());
-        if (binaryHash != null) {
-            // binaryHash will be null if the React Native assets were not bundled into the APK
-            // (e.g. in Debug builds)
-            configMap.putString(CodePushConstants.PACKAGE_HASH_KEY, binaryHash);
+
+        Activity currentActivity = getCurrentActivity();
+        if (currentActivity != null) {
+            String binaryHash = CodePushUpdateUtils.getHashForBinaryContents(currentActivity, mCodePush.isDebugMode());
```

Comment:
I think we should update `getHashForBinaryContents` to use a Context instead of an Activity. Really, all its doing is looking in the assets folder to retrieve the hash which we computed during compile time. So it doesn't need an Activity, we could use the `reactContext`.

In addition, since the binary hash is precomputed during compile time and is rather cheap to just retrieve it, maybe it would be okay to retrieve it during app start like what you are doing for the clientUniqueId. But I can see your apprehension to adding additional steps on app start, so maybe this is less important.